### PR TITLE
CMR-7088: update API routes and documentation for bulk granule update

### PR DESCRIPTION
**Pull request by mschmele**
Laying more foundation for bulk granule update. The majority of the work for this feature is going to be done in the upcoming sprint

### CODE REVIEW

**Comment by mschmele** [comment]
I have more updates that I need to make to the documentation

File: ingest-app/src/cmr/ingest/api/bulk.clj [hunk 1]
```diff
@@ -34,6 +34,22 @@
           {:status 200
            :task-id task-id})))))
 
+(defn bulk-update-granules
+  "Bulk update granules. Validatie prvider exists, check ACLs, and validate
```
Comment (jaybarra) [inline ingest-app/src/cmr/ingest/api/bulk.clj:38]: typos, `Validate` and 'provider'` are misspelled

File: ingest-app/src/cmr/ingest/api/bulk.clj [hunk 2]
```diff
@@ -34,6 +34,22 @@
...
+   POST body. Writes tasks to queue for worker to begin processing."
+  [provider-id request]
+  (if-not (ingest-config/granule-bulk-update-enabled)
```
Comment (jaybarra) [inline ingest-app/src/cmr/ingest/api/bulk.clj:41]: Should update `granule-bulk-update-enabled` name to end with `?`

File: ingest-app/src/cmr/ingest/api/bulk.clj [hunk 3]
```diff
@@ -34,6 +34,22 @@
...
+    (srvc-errors/throw-service-error :bad-request "Bulk update is disabled")
```
Comment (jaybarra) [inline ingest-app/src/cmr/ingest/api/bulk.clj:42]: May want to specify this bulk update is for granules

**Review by jaybarra — CHANGES_REQUESTED** [review]

File: ingest-app/src/cmr/ingest/api/bulk.clj (hunk 2 above)
Comment (mschmele) [reply ingest-app/src/cmr/ingest/api/bulk.clj:41]: The other config value didn't end with `?`, but I like that better

File: ingest-app/src/cmr/ingest/services/granule_bulk_update_service.clj [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+   [cmr.transmit.metadata-db :as mdb]
+   [cmr.transmit.metadata-db2 :as mdb2]
+   [cmr.umm-spec.field-update :as field-update]))
+
+(defn validate-and-save-bulk-granule-update
+ "Validate the bulk update POST parameters, save rows to the db for task
+ and granule statuses, and queue bulk granule update. Return task id, which comes
+ from the db save."
+ [context provider-id json user-id]
+ ;; (validate-bulk-update-post-params json)
```
Comment (jaybarra) [inline ingest-app/src/cmr/ingest/services/granule_bulk_update_service.clj:24]: Removed the commented out section

**Review by jaybarra — APPROVED** [review]

File: ingest-app/src/cmr/ingest/config.clj [hunk 5]
```diff
@@ -12,8 +12,12 @@
...
 
-(defconfig bulk-update-enabled
-  "Flag for whether or not bulk update is enabled."
+(defconfig collection-bulk-update-enabled
+  "Flag for whether or not bulk collection update is enabled."
+  {:default true :type Boolean})
+
+(defconfig granule-bulk-update-enabled?
+  "Flag for whether or not bulk granule update is enabled."
   {:default true :type Boolean})
```
Comment (yiduoke) [inline ingest-app/src/cmr/ingest/config.clj:21]: ```collection-bulk-update-enabled``` should also have question mark at the end

**Review by yiduoke — CHANGES_REQUESTED** [review]

**Review by yiduoke — APPROVED** [review]

**Review by jceaser — APPROVED** [review]
my suggestion was considered.

**Pull request merged**